sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/persons/PersonDirectorySteps.java

```diff
@@ -106,25 +211,83 @@ public PersonDirectorySteps(
                       + " "
                       + apiState.getLastCreatedPerson().getFirstName();
               break;
-              // etc
+            case "phone number":
+              searchText = apiState.getLastCreatedPerson().getPhone();
+              break;
+            case "email":
+              searchText = apiState.getLastCreatedPerson().getEmailAddress();
+              break;
           }
-          webDriverHelpers.waitUntilElementIsVisibleAndClickable(APPLY_FILTERS_BUTTON);
-          webDriverHelpers.clickOnWebElementBySelector(ALL_BUTTON);
-          webDriverHelpers.waitForPageLoadingSpinnerToDisappear(150);
 
           webDriverHelpers.fillInWebElement(MULTIPLE_OPTIONS_SEARCH_INPUT, searchText);
+        });
+
+    Then(
+        "I change id,full name,email,phone number, by {string}",
+        (String searchCriteria) -> {
+          String searchText = "";
+          String personUUID =
+              dataOperations.getPartialUuidFromAssociatedLink("XCBJ2T-XYUGE3-QZK6PD-CWL6KBER");
+          switch (searchCriteria) {
+            case "uuid":
+              searchText = personUUID;
+              break;
+            case "full name":
+              searchText = "Tom Jerry";
+              break;
+            case "phone number":
+              searchText = "(06713) 6606268";
+              break;
+            case "email":
+              searchText = "Tom.Jerry@person.com";
+              break;
+          }
+          webDriverHelpers.fillInWebElement(MULTIPLE_OPTIONS_SEARCH_INPUT, searchText);
+        });
+
+    When(
+        "I navigate to the last created Person page via URL",
```

Comment:
add also from where was the person created since a similar method already exists